### PR TITLE
HW5: MongoDB

### PR DESCRIPTION
**Pull request by Myxjiew**
Designed MondoDB structure;
Seeded fake data;

### CODE REVIEW

File: api/models/post.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    required: true,
+  },
+
+  content: {
+    type: String,
+    required: true,
+  },
+
+  likes: {
+    type: Number,
```
Comment (afandul) [inline api/models/post.js]: Could be ref to users that liked this post

File: api/models/post.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  comments: [
+    {
+      type: Schema.Types.ObjectId,
+      ref: "Comment",
+    },
+  ],
+
+  tags: [
+    {
+      type: Schema.Types.ObjectId,
```
Comment (afandul) [inline api/models/post.js]: Could be tag directly instead of id, since tag text is unique

File: api/models/tag.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const likeSchema = Schema({
+  tagName: {
+    type: String,
+    required: true,
+  },
+
+  post: [
```
Comment (afandul) [inline api/models/tag.js]: Should be `posts`

File: api/models/tag.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const likeSchema = Schema({
```
Comment (afandul) [inline api/models/tag.js]: Should be `tagSchema`

File: api/models/user.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  login: {
+    type: String,
+    required: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  favorites: {
+    type: [String],
```
Comment (afandul) [inline api/models/user.js]: If that's about liked posts, better ref to Likes to Posts

File: api/models/comment.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const commentSchema = Schema({
+  text: {
+    type: String,
+    required: true,
+  },
+  author: [
```
Comment (afandul) [inline api/models/comment.js]: Multiple authors under single comment? Rather multiple comments from single author

File: api/models/comment.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      type: Schema.Types.ObjectId,
+      ref: "User",
+      required: true,
+    },
+  ],
+  post: [
```
Comment (afandul) [inline api/models/comment.js]: Single comment under multiple posts? Rather multiple comments under single post

File: api/db_connection.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require("mongoose");
+
+async function connect() {
+  try {
+    const connection = await mongoose.connect(
+      "mongodb+srv://admin:admin@bci-mongodb.ru25h.mongodb.net/BCI-MongoDB?retryWrites=true&w=majority"
```
Comment (afandul) [inline api/db_connection.js]: Keep it secret in env and remove from SCM asap

File: api/controllers/tags.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+const { Tag } = require("../models/tag.js");
+
+async function getAllTags() {
+  return Tag.find({});
+}
+
+async function addTag(data) {
+  const tag = new Tag({
+    tagName: data.tagName,
+    post: [],
```
Comment (afandul) [inline api/controllers/tags.js]: What about creating tag with existing posts? Keep it as default for optional value`posts: req.body.posts || []`

File: api/controllers/users.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+async function updateUser(id, dataUser) {
+  await User.findByIdAndUpdate(id, dataUser);
+  return {
+    status: 200,
+  };
+}
+
+async function removeUser(id) {
+  await User.remove({ _id: id });
```
Comment (afandul) [inline api/controllers/users.js]: Could chain these two into transaction

File: api/routes/comments.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+const {
+  getAllComments,
+  addComment,
+  removeComment,
+  updateComment,
+} = require("../controllers/comments.js");
+
+router.get("/", async (req, res) => {
+  res.send(await getAllComments());
```
Comment (afandul) [inline api/routes/comments.js]: Better keep await operations in separate line

**Review by afandul — APPROVED** [review]
Solid solution. Pay attention to entity relations and missing like functionality.

File: api/models/post.js (hunk 1 above)
Comment (Myxjiew) [reply api/models/post.js]: Changed

File: api/models/post.js (hunk 2 above)
Comment (Myxjiew) [reply api/models/post.js]: Fixed

File: api/models/tag.js (hunk 3 above)
Comment (Myxjiew) [reply api/models/tag.js]: Fixed

File: api/models/tag.js (hunk 4 above)
Comment (Myxjiew) [reply api/models/tag.js]: Fixed

File: api/db_connection.js (hunk 8 above)
Comment (Myxjiew) [reply api/db_connection.js]: Changed

File: api/models/comment.js (hunk 6 above)
Comment (Myxjiew) [reply api/models/comment.js]: Fixed

File: api/models/comment.js (hunk 7 above)
Comment (Myxjiew) [reply api/models/comment.js]: Fixed

File: api/controllers/tags.js (hunk 9 above)
Comment (Myxjiew) [reply api/controllers/tags.js]: Fixed

File: api/models/user.js (hunk 5 above)
Comment (Myxjiew) [reply api/models/user.js]: Fixed

File: api/routes/comments.js (hunk 11 above)
Comment (Myxjiew) [reply api/routes/comments.js]: Changed

File: api/controllers/users.js (hunk 10 above)
Comment (Myxjiew) [reply api/controllers/users.js]: Fixed

**Pull request merged**